### PR TITLE
Flutter - Add better viewInfoExtractor documentation

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What does this PR do?

Added some clarification that `eventMappers` shouldn't be used for view renaming, and that `viewInfoExtractor`s should be used instead.  Also added more documentation to the index page about the viewInfoExtractor parameter.

### Motivation
Some support requests recently brought to my attention that the viewInfoExtractor already existed and is the proper method for renaming views. eventMappers can cause weird issues if you don't rename views on all the events associated with a view.

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/real_user_monitoring/flutter/advanced_configuration.md [hunk 1]
```diff
@@ -134,6 +134,8 @@ DatadogSdk.instance.setUserInfo("1234", "John Doe", "john@doe.com");
 
 ## Modify or drop RUM events
 
+**Note**: This feature is not yet available for Flutter Web applications.
```
Comment (urseberry) [inline content/en/real_user_monitoring/flutter/advanced_configuration.md:137]:
```suggestion
**Note**: This feature is not yet available for Flutter web applications.
```

File: content/en/real_user_monitoring/flutter/advanced_configuration.md [hunk 2]
```diff
@@ -169,24 +171,26 @@ Depending on the event's type, only some specific properties can be modified:
...
+|                  | `errorEvent.view.name`            | Name the view linked to this action.¹         |
 |                  | `errorEvent.view.referrer`        | Referrer of the view linked to this action.   |
 |                  | `errorEvent.view.url`             | URL of the view linked to this error.         |
 | RumResourceEvent | `resourceEvent.resource.url`      | URL of the resource.                          |
-|                  | `resourceEvent.view.name`         | Name the view linked to this action.          |
+|                  | `resourceEvent.view.name`         | Name the view linked to this action.¹         |
 |                  | `resourceEvent.view.referrer`     | Referrer of the view linked to this action.   |
 |                  | `resourceEvent.view.url`          | URL of the view linked to this resource.      |
 
+¹While event mappers allow you to modify the names of view, but this is not the recommended way to rename a view. Use the `viewInfoExtractor` parameter on [`DatadogNavigationObserver`][7] instead
```
Comment (urseberry) [inline content/en/real_user_monitoring/flutter/advanced_configuration.md:192]:
```suggestion
¹ While event mappers allow you to modify the names of views, they are not the recommended way to rename a view. Use the `viewInfoExtractor` parameter on [`DatadogNavigationObserver`][7] instead
```

File: content/en/real_user_monitoring/flutter/_index.md [hunk 3]
```diff
@@ -50,11 +50,33 @@ MaterialApp(
 
 This works if you are using named routes or if you have supplied a name to the `settings` parameter of your `PageRoute`.
 
-Alternatively, you can use the `DatadogRouteAwareMixin` property in conjunction with the `DatadogNavigationObserverProvider` property to start and stop your RUM views automatically. With `DatadogRouteAwareMixin`, move any logic from `initState` to `didPush`. 
+Alternatively, you can use the `DatadogRouteAwareMixin` property in conjunction with the `DatadogNavigationObserverProvider` property to start and stop your RUM views automatically. With `DatadogRouteAwareMixin`, move any logic from `initState` to `didPush`.
+
+To rename your views or supply custom paths, provide a [`viewInfoExtractor`][8] callback. This function can fall back to the default behavior of the observer by calling `defaultviewInfoExtractor`. For example:
```
Comment (0xnm) [inline content/en/real_user_monitoring/flutter/_index.md:55]:
```suggestion
To rename your views or supply custom paths, provide a [`viewInfoExtractor`][8] callback. This function can fall back to the default behavior of the observer by calling `defaultViewInfoExtractor`. For example:
```

**Review by 0xnm — APPROVED** [review]
added a typo fix suggestion

**Review by plousada — APPROVED** [review]

**Review by urseberry — APPROVED** [review]

**Pull request merged**